helix-core/src/main/java/org/apache/helix/manager/zk/zookeeper/ZkClient.java

```diff
@@ -145,32 +145,28 @@ public int hashCode() {
 
   private class ZkPathStatRecord {
     private final String _path;
-    private Stat _stat = null;
-    private boolean _checked = false;
+    private AtomicBoolean _isChecked = new AtomicBoolean(false);
```

Comment:
Can you please explain why we need to change this value to Atomic?

Reply:
It was my simple try to make it thread-safe, the atomic has the ability though; Do you have concrete suggestions on how to use it?

Reply:
But is it really necessary to use atomic? Is there a clue that there would be concurrency issue?

Reply:
@pkuwm I agree it's not necessary (my understanding of this code currently involves no multi-threading; To make it thread-safe with a simple atomic boolean is merely a simple try. 
@jiajunwang @pkuwm The latest updates involve little changes on the existing code, it's no longer needed plus I don't really want to improve (refactor) the existing ones :)